src/workloads/query/TenMDocCollection_IntId.yml

```diff
@@ -51,7 +62,7 @@ ActorTemplates:
         Operations:
         - OperationName: find
           OperationCommand:
-            Filter: {_id: {$in: {^FixedGeneratedValue: {fromGenerator: {^Array: {distinct: true, number: 100000, of: {^RandomInt: {min: 0, max: 9999999}}}}}}}}
+            Filter: {_id: {$in: [{^RandomInt: {min: 0, max: 9999999}}]}}
```

Comment:
I see you trimmed the number of in values. Is there a story here? Does this single $in get optimized into an equality? If yes, no need to keep it I don't think. If not, then can we add to it?

In either case, I think we would benefit from a medium-sized in list and a large size in list. I've seen some large ones, including in a recent help ticket where they are causing problems. I don't really know what large means exactly, but 10k seems good? medium is maybe 1k then?

Reply:
Ah... yea, a length of 1 makes it an `$eq`. But a length of 2 keeps it as an `$in`.

I'm still confused about what the purpose of testing with an `$in` of 10K points is. This is a fundamentally different query than the IDHACK query.

Reply:
Although, funnily, `$eq` doesn't invoke `IDHACK`...